rclcpp/include/rclcpp/publisher_base.hpp

```diff
@@ -203,6 +205,46 @@ class PublisherBase : public std::enable_shared_from_this<PublisherBase>
   std::vector<rclcpp::NetworkFlowEndpoint>
   get_network_flow_endpoints() const;
 
+  /// Wait until all published messages are acknowledged or until the specified timeout elapses.
+  /**
+   * This method waits until all published messages are acknowledged by all matching
+   * subscriptions or the given timeout elapses.
+   *
+   * If the timeout is negative then this method will block indefinitely until all published
+   * message data are acknowledged.
```

Comment:
```suggestion
   * messages are acknowledged.
```

I think you can either say "messages are" or "message data is".
My preference is the first option.
I might be wrong, as I'm not a native speaker.

Reply:
Agree.  
I'm not a native speaker too :)